src/cmap/auth/mongodb_oidc.ts

```diff
@@ -0,0 +1,26 @@
+/** @public */
+export interface OIDCMechanismServerStep1 {
+  authorizeEndpoint?: string;
+  tokenEndpoint?: string;
+  deviceAuthorizeEndpoint?: string;
+  clientId: string;
+  clientSecret?: string;
+  requestScopes?: string[];
+}
+
+/** @public */
+export interface OIDCRequestTokenResult {
+  accessToken: string;
+  expiresInSeconds?: number;
+  refreshToken?: string;
+}
+
+/** @public */
+export interface OIDCRequestFunction {
+  (idl: OIDCMechanismServerStep1): Promise<OIDCRequestTokenResult>;
+}
+
+/** @public */
+export interface OIDCRefreshFunction {
+  (idl: OIDCMechanismServerStep1, result: OIDCRequestTokenResult): Promise<OIDCRequestTokenResult>;
+}
```

Comment:
I think the only reason to do a callable interface is to support overloads which we could always do in the future, but for now I think defining a strict type suffices:
```suggestion
export type OIDCRefreshFunction = (idl: OIDCMechanismServerStep1, result: OIDCRequestTokenResult) => Promise<OIDCRequestTokenResult>;
```

And same for the other callback type.

Reply:
Changed to strict types.